pkg/opentracing/opentracing.go

```diff
@@ -95,11 +97,43 @@ func (t *Tracing) buildJaeger(componentName string, c config.JaegerTracing) (ope
 		},
 	}
 
-	return cfg.New(
-		componentName,
-		jaegercfg.Logger(jaegerLoggerAdapter{log.StandardLogger()}),
-		jaegercfg.Metrics(metrics.NullFactory),
+	tracerMetrics := jaeger.NewMetrics(metrics.NullFactory, nil)
+	tracerLogger := jaegerLoggerAdapter{log.StandardLogger()}
+	sampler, err := cfg.Sampler.NewSampler(componentName, tracerMetrics)
+	if err != nil {
+		log.WithError(err).Error("failed to create jaeger sampler")
```

Comment:
`sampler` creation failed, but you're not returning from the function and trying to use broken sampler bellow